File: ts/project/.sdk/tm/rb/utility/make_response.rb
# ProjectName SDK utility: make_response
module ProjectNameUtilities
  MakeResponse = ->(ctx) {
    return ctx.out["response"], nil if ctx.out["response"]
    utility = ctx.utility
    spec = ctx.spec
    result = ctx.result
    response = ctx.response

    return nil, ctx.make_error("response_no_spec", "Expected context spec property to be defined.") unless spec
    return nil, ctx.make_error("response_no_response", "Expected context response property to be defined.") unless response
    return nil, ctx.make_error("response_no_result", "Expected context result property to be defined.") unless result

    spec.step = "response"
    utility.result_basic.call(ctx)
    utility.result_headers.call(ctx)
    utility.result_body.call(ctx)
    utility.transform_response.call(ctx)

    result.ok = true if result.err.nil?
    ctx.ctrl.explain["result"] = result if ctx.ctrl.explain

    return response, nil
  }
end

File: ts/project/.sdk/tm/php/utility/MakeResponse.php
<?php
declare(strict_types=1);

// ProjectName SDK utility: make_response

class ProjectNameMakeResponse
{
    public static function call(ProjectNameContext $ctx): array
    {
        if (isset($ctx->out['response'])) {
            return [$ctx->out['response'], null];
        }
        $utility = $ctx->utility;
        $spec = $ctx->spec;
        $result = $ctx->result;
        $response = $ctx->response;

        if (!$spec) {
            return [null, $ctx->make_error('response_no_spec', 'Expected context spec property to be defined.')];
        }
        if (!$response) {
            return [null, $ctx->make_error('response_no_response', 'Expected context response property to be defined.')];
        }
        if (!$result) {
            return [null, $ctx->make_error('response_no_result', 'Expected context result property to be defined.')];
        }

        $spec->step = 'response';
        ($utility->result_basic)($ctx);
        ($utility->result_headers)($ctx);
        ($utility->result_body)($ctx);
        ($utility->transform_response)($ctx);

        if ($result->err === null) {
            $result->ok = true;
        }
        if ($ctx->ctrl->explain) {
            $ctx->ctrl->explain['result'] = $result;
        }

        return [$response, null];
    }
}
